hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionOperatorImpl.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.eviction;
+
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MaxSizeConfig;
+import com.hazelcast.core.EntryEventType;
+import com.hazelcast.map.impl.MapServiceContext;
+import com.hazelcast.map.impl.event.MapEventPublisher;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.map.impl.recordstore.RecordStore;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.EventService;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.util.Clock;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+import static com.hazelcast.cluster.memberselector.MemberSelectors.DATA_MEMBER_SELECTOR;
+import static com.hazelcast.core.EntryEventType.EVICTED;
+import static com.hazelcast.core.EntryEventType.EXPIRED;
+import static com.hazelcast.map.impl.MapService.SERVICE_NAME;
+
+/**
+ * Eviction helper methods.
+ */
+public class EvictionOperatorImpl implements EvictionOperator {
```

Comment:
This is clearly not an operation since it doesn't extend Operation. So therefor it should not be called Operation. This makes code very difficult to understand because assumptions are violated. Good code can be read as a story and somebodies assumptions should not be violated (too often).